cmd/cmd_buld.go

```diff
@@ -0,0 +1,109 @@
+package cmd
+
+import (
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/dynamic"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/types/descriptorpb"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+func NewBuildCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "build <MessageName>",
+		Short: "Build json by proto message",
+		Args:  messageNameRequired,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			// parse protofiles & create proto object
+			p, err := parseProtofiles()
+			if err != nil {
+				return
+			}
+
+			messageDescriptor, err := findMessage(p, args[0])
+			if err != nil {
+				return
+			}
+
+			msg := buildMessage(dynamic.NewMessage(messageDescriptor))
+
+			b, err := msg.MarshalJSONPB(&jsonpb.Marshaler{
+				EmitDefaults: true,
+				Indent:       "  ",
+			})
+			if err != nil {
+				return
+			}
+
+			cmd.Println(string(b))
+
+			return
+		},
+	}
+
+	return cmd
+}
+
+func buildMessage(message *dynamic.Message) *dynamic.Message {
+	for _, field := range message.GetMessageDescriptor().GetFields() {
+		if field.IsRepeated() {
+			message.SetField(field, []interface{}{buildDefaultValue(field)})
+		} else if field.IsMap() {
+			message.SetField(
+				field,
+				map[interface{}]interface{}{
+					buildDefaultValue(field.GetMapKeyType()): buildDefaultValue(field.GetMapValueType()),
+				},
+			)
+		} else if field.GetOneOf() != nil {
+			oneOfField := field.GetOneOf().GetChoices()[0]
+			message.SetField(oneOfField, buildDefaultValue(oneOfField))
+		} else {
+			message.SetField(field, buildDefaultValue(field))
+		}
+	}
```

Comment:
Можно немного сократить:
```suggestion
		switch {
		case field.IsRepeated():
			message.SetField(field, []interface{}{buildDefaultValue(field)})
		case field.IsMap():
			message.SetField(
				field,
				map[interface{}]interface{}{
					buildDefaultValue(field.GetMapKeyType()): buildDefaultValue(field.GetMapValueType()),
				},
			)
		case field.GetOneOf() != nil:
			oneOfField := field.GetOneOf().GetChoices()[0]
			message.SetField(oneOfField, buildDefaultValue(oneOfField))
		default:
			message.SetField(field, buildDefaultValue(field))
		}
	}
```

Reply:
Тут просто ифы заменены на свич? А зачем?

Reply:
Это чисто стилевой момент, получается немного компактнее. Вообще в конфигурации golangci-lint включен go-critic, он такое подсвечивает: https://go-critic.com/overview#ifelsechain. В целом можно оставить как есть, не критично:)

Reply:
Понял. Принял)